staging/src/k8s.io/client-go/tools/portforward/portforward.go

```diff
@@ -117,28 +117,35 @@ type listenAddress struct {
 	failureMode string
 }
 
+// parseAddresses parses, validates addresses
+// Keeps the order from addressesToParse, but removes duplicates
 func parseAddresses(addressesToParse []string) ([]listenAddress, error) {
-	var addresses []listenAddress
+	var keyOrder []string
 	parsed := make(map[string]listenAddress)
 	for _, address := range addressesToParse {
 		if address == "localhost" {
 			ip := listenAddress{address: "127.0.0.1", protocol: "tcp4", failureMode: "all"}
 			parsed[ip.address] = ip
+			// IPv4 address for localhost must go first in keyOrder for backward compatibility
```

Comment:
the issue raised in https://github.com/kubernetes/kubernetes/issues/71017 is legitimate. I think it might be as simple as ensuring we never switch an existing "any" rule to an "all" rule

Reply:
> I don't recall this being a backwards compatibility requirement.

Sorry, for the confusion again. Initially I was looking into flaky tests and though older tests (skew) expect `127.0.0.1` to go first in logs, but then I realised that tests check log output with regexp and that it's not relevant.

I can remove this comment to avoid further confusion.

Reply:
@m1kola yes please remove the comment and add a test verifying the issue mentioned is fixed too.